backend/apps/project_health/services.py

```
@@ -0,0 +1,216 @@
+def analyze_repository(repo_url: str, token: str = None) -> dict:
+    """
+    Fetch GitHub data, run sentiment analysis, compute health score.
+    Returns a dict of all metrics ready to store in RepoHealthScore.
+    """
+    owner, repo_name = _parse_repo_url(repo_url)
+
+    # Fetch basic repo info
+    try:
+        repo_data = _github_get(f"/repos/{owner}/{repo_name}", token=token)
+    except requests.HTTPError as e:
+        raise ValueError(f"Repository not found or inaccessible: {e}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)